code/game/objects/items/storage/backpack.dm

```diff
@@ -365,6 +365,64 @@
 	icon_state = "smock"
 	storage_type = /datum/storage/backpack/no_delay
 
+/obj/item/storage/backpack/marine/duffelbag
+	name = "\improper TGMC Duffelbag"
+	desc = "A massive bag designed for those interested in being the squad Mule. \
+	Features extra storage space at the cost of speed. \
+	Any squadmates can easily access the storage with right-click"
+	icon_state = "duffel"
+	worn_icon_state = "duffel"
+	storage_type = /datum/storage/backpack/duffelbag
+	slowdown = 0.2
+
+/obj/item/storage/backpack/marine/duffelbag/equipped(mob/equipper, slot)
+	. = ..()
+	if(slot == SLOT_BACK)
+		RegisterSignal(equipper, COMSIG_CLICK_RIGHT, PROC_REF(on_rclick_duffel_wearer))
+		for(var/mob/M AS in storage_datum.content_watchers)
+			storage_datum.close(M)
+
+/obj/item/storage/backpack/marine/duffelbag/unequipped(mob/unequipper, slot)
+	. = ..()
+	UnregisterSignal(unequipper, COMSIG_CLICK_RIGHT)
+
+///Allows non-wearers to access this inventory
+/obj/item/storage/backpack/marine/duffelbag/proc/on_rclick_duffel_wearer(datum/source, mob/clicker)
+	SIGNAL_HANDLER
+	if(clicker == loc) //Wearer can't open this inventory
+		return
+	storage_datum.open(clicker)
+
+/datum/storage/backpack/duffelbag
+	max_storage_space = 48
+	access_delay = 0
+
+/datum/storage/backpack/duffelbag/put_storage_in_hand(datum/source, obj/over_object, mob/living/carbon/human/user)
+	//Taking off the duffelbag has a channel
+	if(user.back != parent || !do_after(user, 3 SECONDS))
+		return COMPONENT_NO_MOUSEDROP
+
+	switch(over_object.name)
+		if("r_hand")
+			INVOKE_ASYNC(src, PROC_REF(put_item_in_r_hand), source, user)
+		if("l_hand")
+			INVOKE_ASYNC(src, PROC_REF(put_item_in_l_hand), source, user)
+	return COMPONENT_NO_MOUSEDROP
+
+/datum/storage/backpack/duffelbag/open(mob/user)
+	if(!iscarbon(user))
+		return TRUE
+	var/mob/living/carbon/carbon_user = user
+	if(carbon_user == parent.loc && carbon_user.back == parent)
```

Comment:
do you actually need both checks here?

Reply:
Nope. I just need to check if it's being currently worn by the person right-clicking the wearer.